### PR TITLE
TE5-T375 Android UI Task Completed

### PR DESCRIPTION
**Pull request by jay-a-shah**

### CODE REVIEW

File: app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import android.graphics.Color
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.Gravity
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+
+class BasicsKotlin : AppCompatActivity() {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt]: Space before colon not required

File: app/src/main/res/layout/activity_android_ui.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_android_ui.xml:2]: Code formatting

File: app/src/main/res/layout/activity_basics_kotlin.xml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        android:fontFamily="sans-serif-black"
+        android:gravity="center"
+        android:text="Registration Form"
+        android:textColor="@color/purple"
+        android:textSize="30dp"
+        app:layout_constraintBottom_toTopOf="@id/editTextTextPersonName"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+    <EditText
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_basics_kotlin.xml]: Code formatting in whole file

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+    <string name="radiobtn_male">Male</string>
+    <string name="radiobtn_female">Female</string>
+    <string name="textview_pref">Your Prefrences</string>
+    <string name="checkbox_kotlin">Kotlin</string>
+    <string name="checkbox_ios">IOS</string>
+    <string name="checkbox_angular">Angular Js</string>
+    <string name="btn_submit">Submit</string>
+    <string name="customToastText">All Done</string>
+    <string name="android_Ui_Button">New Button</string>
+    <string name="Basic_kotlin_gender_textview">Gender:-</string>
```
Comment (purvesh-d) [inline app/src/main/res/values/strings.xml]: follow same naming convention , underscore not camel case

**Review by purvesh-d — CHANGES_REQUESTED** [review]
- Code formatting is required in all the files

File: app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt [hunk 5]
```diff
@@ -1,16 +1,36 @@
...
     }
+
+    fun showToast() {
+        val toastView = layoutInflater.inflate(
+            R.layout.custom_toast, findViewById(R.id.customToastLayout)
+        )
+        val myToast = Toast(applicationContext)
+        myToast.duration = Toast.LENGTH_SHORT
+        myToast.setGravity(Gravity.CENTER_VERTICAL, 0, 0)
+        myToast.view = toastView//setting the view of custom toast layout
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt:33]: Create constant file and move static integer there

File: app/src/main/java/com/example/demoappkotlin/MainActivity.kt [hunk 6]
```diff
@@ -11,6 +14,24 @@ class MainActivity : AppCompatActivity() {
...
+            startActivity(intent)
+        }
+        //Android Ui Button with Normal toast
+        val androiduibtn = findViewById<Button>(R.id.android_ui_btn)
+        androiduibtn.setOnClickListener {
+            val normalToast = Toast.makeText(
+                applicationContext, "Moving Towards new Activity",
+                Toast.LENGTH_SHORT
+            )
+            normalToast.setGravity(Gravity.CENTER_VERTICAL, 0, 0)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/MainActivity.kt]: Same comment for static 0

File: app/src/main/java/com/example/demoappkotlin/MainActivity.kt [hunk 7]
```diff
@@ -11,6 +14,24 @@ class MainActivity : AppCompatActivity() {
...
+        val nextbutton = findViewById<Button>(R.id.button)
+        nextbutton.setOnClickListener {
+            val intent = Intent(applicationContext, BasicsKotlin::class.java)
+            startActivity(intent)
+        }
+        //Android Ui Button with Normal toast
+        val androiduibtn = findViewById<Button>(R.id.android_ui_btn)
+        androiduibtn.setOnClickListener {
+            val normalToast = Toast.makeText(
+                applicationContext, "Moving Towards new Activity",
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/MainActivity.kt]: Move string to string file

File: app/src/main/res/layout/activity_basics_kotlin.xml [hunk 8]
```diff
@@ -4,6 +4,215 @@
...
+        android:inputType="textPersonName"
+        android:paddingTop="12dp"
+        android:paddingBottom="12dp"
+        android:singleLine="false"
+        android:textColorHint="@color/purple"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/Textview" />
+
+    <EditText
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_basics_kotlin.xml]: Use AppCompact edittext

File: app/src/main/res/layout/activity_basics_kotlin.xml [hunk 9]
```diff
@@ -4,6 +4,215 @@
...
+        android:inputType="phone"
+        android:singleLine="true"
+        android:textAlignment="center"
+        android:textColorHint="@color/purple"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.497"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/editTextEmailAddress" />
+
+    <TextView
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_basics_kotlin.xml]: Use only appCompact/androidx widgets in whole file

File: app/src/main/res/layout/activity_main.xml [hunk 10]
```diff
@@ -4,26 +4,125 @@
...
+        app:layout_constraintHorizontal_bias="0.265"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/imageView"
+        tools:ignore="MissingConstraints">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="vertical" />
+    </ScrollView>
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: need empty line

File: app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -4,26 +4,125 @@
...
+        android:textSize="16sp"
+        android:textStyle="bold"
+        android:thumbTint="#501010"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.883"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="UseSwitchCompatOrMaterialXml" />
+
+    <CheckBox
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_main.xml]: Use appcompcat widgets

File: app/src/main/res/layout/custom_toast.xml [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    <androidx.appcompat.widget.AppCompatTextView
+        android:id="@+id/customtextview"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_margin="5dp"
+        android:text="@string/customToastText"
+        android:textColor="@color/custom_toast_text"
+        android:textSize="38sp"
+        android:textStyle="bold"
+        app:layout_constraintStart_toEndOf="@id/imageView_custom_toast" />
```
Comment (purvesh-d) [inline app/src/main/res/layout/custom_toast.xml:29]: need empty line

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt (hunk 5 above)
Comment (jay-a-shah) [reply app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt:33]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 10 above)
Comment (jay-a-shah) [reply app/src/main/res/layout/activity_main.xml]: Okk

File: app/src/main/res/layout/custom_toast.xml (hunk 12 above)
Comment (jay-a-shah) [reply app/src/main/res/layout/custom_toast.xml:29]: Okk

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (jay-a-shah) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/main/res/layout/activity_basics_kotlin.xml [hunk 13]
```diff
@@ -4,6 +4,215 @@
...
+        android:text="@string/Basic_kotlin_gender_textview"
+        android:textColor="@color/purple"
+        android:textSize="25sp"
+        android:textStyle="bold"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.123"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/editTextPhone" />
+
+    <RadioGroup
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_basics_kotlin.xml:91]: use appcompact widgets

**Review by purvesh-d — CHANGES_REQUESTED** [review]

**Pull request merged**